File: AssetManager/Helpers/ExtendedMethods.vb
Imports System.Data.Common
Imports System.Reflection

Module ExtendedMethods

    Public Sub DoubleBufferedDataGrid(ByVal dgv As DataGridView, ByVal setting As Boolean)
        Dim dgvType As Type = dgv.[GetType]()
        Dim pi As PropertyInfo = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance Or BindingFlags.NonPublic)
        pi.SetValue(dgv, setting, Nothing)
    End Sub

    Public Sub DoubleBufferedListView(ByVal dgv As ListView, ByVal setting As Boolean)
        Dim dgvType As Type = dgv.[GetType]()
        Dim pi As PropertyInfo = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance Or BindingFlags.NonPublic)
        pi.SetValue(dgv, setting, Nothing)
    End Sub

    Public Sub DoubleBufferedListBox(ByVal dgv As ListBox, ByVal setting As Boolean)
        Dim dgvType As Type = dgv.[GetType]()
        Dim pi As PropertyInfo = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance Or BindingFlags.NonPublic)
        pi.SetValue(dgv, setting, Nothing)
    End Sub

    Public Sub DoubleBufferedFlowLayout(ByVal dgv As FlowLayoutPanel, ByVal setting As Boolean)
        Dim dgvType As Type = dgv.[GetType]()
        Dim pi As PropertyInfo = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance Or BindingFlags.NonPublic)
        pi.SetValue(dgv, setting, Nothing)
    End Sub

    Public Sub DoubleBufferedTableLayout(ByVal dgv As TableLayoutPanel, ByVal setting As Boolean)
        Dim dgvType As Type = dgv.[GetType]()
        Dim pi As PropertyInfo = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance Or BindingFlags.NonPublic)
        pi.SetValue(dgv, setting, Nothing)
    End Sub

    Public Sub DoubleBufferedPanel(ByVal dgv As Panel, ByVal setting As Boolean)
        Dim dgvType As Type = dgv.[GetType]()
        Dim pi As PropertyInfo = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance Or BindingFlags.NonPublic)
        pi.SetValue(dgv, setting, Nothing)
    End Sub

    ''' <summary>
    ''' Adds a parameter to the command.
    ''' </summary>
    ''' <param name="command">
    ''' The command.
    ''' </param>
    ''' <param name="parameterName">
    ''' Name of the parameter.
    ''' </param>
    ''' <param name="parameterValue">
    ''' The parameter value.
    ''' </param>
    ''' <remarks>
    ''' </remarks>
    <System.Runtime.CompilerServices.Extension>
    Public Sub AddParameterWithValue(command As DbCommand, parameterName As String, parameterValue As Object)
        Dim parameter = command.CreateParameter()
        parameter.ParameterName = parameterName
        parameter.Value = parameterValue
        command.Parameters.Add(parameter)
    End Sub

End Module

File: AssetManager/Helpers/ExtendedMethods.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Threading.Tasks;
using System.Data.Common;
using System.Reflection;
namespace AssetManager
{

	static class ExtendedMethods
	{

		public static void DoubleBufferedDataGrid(DataGridView dgv, bool setting)
		{
			Type dgvType = dgv.GetType();
			PropertyInfo pi = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance | BindingFlags.NonPublic);
			pi.SetValue(dgv, setting, null);
		}

		public static void DoubleBufferedListView(ListView dgv, bool setting)
		{
			Type dgvType = dgv.GetType();
			PropertyInfo pi = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance | BindingFlags.NonPublic);
			pi.SetValue(dgv, setting, null);
		}

		public static void DoubleBufferedListBox(ListBox dgv, bool setting)
		{
			Type dgvType = dgv.GetType();
			PropertyInfo pi = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance | BindingFlags.NonPublic);
			pi.SetValue(dgv, setting, null);
		}

		public static void DoubleBufferedFlowLayout(FlowLayoutPanel dgv, bool setting)
		{
			Type dgvType = dgv.GetType();
			PropertyInfo pi = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance | BindingFlags.NonPublic);
			pi.SetValue(dgv, setting, null);
		}

		public static void DoubleBufferedTableLayout(TableLayoutPanel dgv, bool setting)
		{
			Type dgvType = dgv.GetType();
			PropertyInfo pi = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance | BindingFlags.NonPublic);
			pi.SetValue(dgv, setting, null);
		}

		public static void DoubleBufferedPanel(Panel dgv, bool setting)
		{
			Type dgvType = dgv.GetType();
			PropertyInfo pi = dgvType.GetProperty("DoubleBuffered", BindingFlags.Instance | BindingFlags.NonPublic);
			pi.SetValue(dgv, setting, null);
		}

		/// <summary>
		/// Adds a parameter to the command.
		/// </summary>
		/// <param name="command">
		/// The command.
		/// </param>
		/// <param name="parameterName">
		/// Name of the parameter.
		/// </param>
		/// <param name="parameterValue">
		/// The parameter value.
		/// </param>
		/// <remarks>
		/// </remarks>
		//[System.Runtime.CompilerServices.Extension()]
		public static void AddParameterWithValue(this DbCommand command, string parameterName, object parameterValue)
		{
			var parameter = command.CreateParameter();
			parameter.ParameterName = parameterName;
			parameter.Value = parameterValue;
			command.Parameters.Add(parameter);
		}

	}
}
